cmd/cspc-operator/app/handler.go

```diff
@@ -95,6 +95,10 @@ func (c *Controller) syncHandler(key string) error {
 	// TODO: Deep-copy only when needed.
 	cspcGot := cspc.DeepCopy()
 	err = c.syncCSPC(cspcGot)
+	if err != nil {
+		return err
+	}
+	err = cleanupCSPIResources(cspcGot)
 	return err
```

Comment:
In the next PR's we can generate an event instead of returning an error. The event will be more visible to the user.

Reply:
Will take this in next PR

Reply:
can you take care of this in same PR?